### DISCUSSION TITLE
Welcome to website discussions!

### DISCUSSION THREAD

**Discussion by SreejithNS** [adrig-technologies/website#6] [General]
Common discussions on website's aspects.

**Comment by SreejithNS** [comment]
**Website Components**

- Navbar - logo, links
- Section - icon, title, subtitle, summary, image, cta
- Section divider - subtitle, summary, icon
- Cards - generic - image, title, caption
- Cards Slider - title, summary, cards
- Quick Cards - Mini cards ( may be variant ) - title, summary, icon
- List of images/cards - Responsive - multiple images or multiple cards
- Footer - groups of links with title

**Specifications for UX:**

- always live ( subtle looping animations ) at every glance
- Themes - Magazine Cover, Creative Story Telling, 
- 3D effects for 2d assets
- sliders for cards that has - image, title and and one line summary
- one section at a time in the view port
- usage of blobs ( unshaped objects )
- 

**Quick Plan:**

1. Start with bare components with static props
2. Make the components dynamic with props as inputs
3. Start with responsive homepage & then Orchestrate components into it, adjust responsive aligns and sizes
4. Start applying colors, rounds, aligns, sizes, responsive behaviours and tune it for perfection.
5. Add real content, tweat the aligns and sizes according the real content
6. Add subtle animations, scroll snaps, intros, loops.
7. Final touches

**Work tips:**

- Communicate/Ask/Get cleared.
- Be clear on what is expected, what is input, what is output.
- Have a visual idea about what you build.
- Develop as an isolated component, build it and finalise it by having broader perspective - Ask yourself where this component will be used in the website and how will it look when I use it in different parts of the website. - for eg. cards
- split commit for individual files if different changes are done to different files - at the same group files together for a single commit if it all has common changes or changed for a common cause.

**Try your best to:**

- not use custom css - use available tailwind classes
- not hardcode variable properties/values
- use Github for disussion - become a pro in collaboration
- document / comment what your code does then there
- use valid variable names - using lengthy variable names is totally fine until it is self explanatory.
- use chatgpt like you know better than it does - be the master not the slave to it
- not panic when full of red lines under your code - try debugging - but if it takes more than 15 mins - reach out for help - we are here to help  - also we are running short on time so decide to reach out soon.

**Reference Websites** - for UI/UX

- https://gsap.com/
- https://www.clubhouse.com/
- https://linktr.ee/
- https://www.realworld.co/
- https://practipago.pe/
- https://keynest.com/

Some Dev References

- https://github.com/saadpasta/developerFolio
- https://github.com/Priyansh42/IT-Company-Website
- https://github.com/darkleas/company-website-reactjs
- https://github.com/Tilhr-Technology/Tilhr-Technology
- https://github.com/learning-zone/website-templates
- https://github.com/imfunniee/fimbo
- https://github.com/codebucks27/The-Weirdos-NFT-Website-Starter-Code
- https://github.com/Sathish2905/sample-website-templates
- https://github.com/mesinkasir/reactwebthemeslandingpage
- https://github.com/web3templates/stablo

**Comment by SreejithNS** [comment]
**Attention**
The branch that you all used for exploring the code and playing around has been **removed**. Kindly use the respective branches created for the issues. 
Also when you initially start working on the issue's branch, follow these steps initially ( only once )

- Delete `node_modules` and `.next` folder
- Pull new changes from main by doing `git pull origin/main`
- Install packages by typing `yarn` in terminal

Reference:
https://github.com/adrig-technologies/website/issues/1#issuecomment-1817977423

**Comment by Vachu0910** [comment]
@SreejithNS package matum remove pannita podhuma or any other changes in the main code na?

**Comment by Vachu0910** [comment]
@SreejithNS changes panniten na once give a look and let me know if anything else need to be done!

**Reply by SreejithNS** [reply]
Changes reviewed and approved 😊
